utils/parenthesesutils.go

```diff
@@ -0,0 +1,123 @@
+package utils
+
+import (
+	"regexp"
+	"sort"
+	"strconv"
+)
+
+type Parentheses struct {
+	OpenIndex  int
+	CloseIndex int
+}
+
+type ParenthesesSlice struct {
+	Parentheses []Parentheses
+}
+
+func (p *ParenthesesSlice) IsPresent(index int) bool {
+	for _, v := range p.Parentheses {
+		if v.OpenIndex == index || v.CloseIndex == index {
+			return true
+		}
+	}
+	return false
+}
+
+func RemovePlaceholderParentheses(pattern, target string) string {
+	parentheses := ParenthesesSlice{getParenthesesIndex(pattern, target)}
+
+	//remove parentheses that have an associate placeholder
```

Comment:
//remove  --> // Remove parentheses which have a corresponding placeholder.
(notice the upper-case R and the leading white-space).

Reply:
Done.